### PR TITLE
[SPARK-47309][SQL] XML: Add schema inference tests for value tags

### PR DESCRIPTION
**Pull request by shujingyang-db**
### What changes were proposed in this pull request?

Add schema inference tags for corrupt records, null values and value tags. For value tags, this PR adds the following tests:
1. Conflict between primitive types conflict
2. Root-level value tag
3. empty value tag in some rows
4. array of value tags:
   1) values split into multiple lines
   2) interspersed in nested structs: empty fields and optional fields in structs
   3) interspersed in arrays and value tags:  empty fields and optional fields in structs
   4) name conflict
   5) CDATA and comments
   6) no spaces / some spaces / whitespaces between valueTags and elements

### Why are the changes needed?

This is a test-only change.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

This is a test-only change.

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlInferSchemaSuite.scala [hunk 1]
```diff
@@ -293,4 +305,323 @@ class XmlInferSchemaSuite extends QueryTest with SharedSparkSession with TestXml
...
+    def makePartition(rdd: RDD[String], parent: File, partName: String, partValue: Any): File = {
+      val p = new File(parent, s"$partName=${partValue.toString}")
+      rdd.saveAsTextFile(p.getCanonicalPath)
+      p
+    }
+
+    withTempPath(root => {
+      withTempView("test_myxml_with_part") {
+        val d1 = new File(root, "d1=1")
+        // root/dt=1/col1=abc
```
Comment (sandip-db) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlInferSchemaSuite.scala:410]: nit:
```suggestion
        // root/d1=1/col1=abc
```

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlInferSchemaSuite.scala [hunk 2]
```diff
@@ -293,4 +305,323 @@ class XmlInferSchemaSuite extends QueryTest with SharedSparkSession with TestXml
...
+        makePartition(
+          sparkContext.parallelize(2 to 5).map(i => s"""<ROW><a>1</a><b>str$i</b></ROW>"""),
+          d1,
+          "col1",
+          "abc"
+        )
+
+        // root/dt=1/col1=abd
```
Comment (sandip-db) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlInferSchemaSuite.scala:418]: nit:
```suggestion
        // root/d1=1/col1=abd
```

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlInferSchemaSuite.scala [hunk 3]
```diff
@@ -293,4 +305,323 @@ class XmlInferSchemaSuite extends QueryTest with SharedSparkSession with TestXml
...
+        "array1",
+        ArrayType(
+          new StructType()
+            .add(valueTagName, ArrayType(StringType))
+            .add(
+              "array2",
+              ArrayType(new StructType()
+                // The value tag is not of long type due to:
+                // When determining
+                .add(valueTagName, ArrayType(StringType))
```
Comment (sandip-db) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlInferSchemaSuite.scala:547]: Update comment.. why it is not LongType?

**Review by sandip-db — APPROVED** [review]

**Review by sandip-db — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**